packages/react-dom/src/events/BrowserEventConstants.js

```diff
@@ -3,95 +3,143 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
  */
 
+import type {TopLevelType} from 'events/TopLevelEventTypes';
+import invariant from 'fbjs/lib/invariant';
+
+import * as DOMTopLevelEventTypes from './DOMTopLevelEventTypes';
 import getVendorPrefixedEventName from './getVendorPrefixedEventName';
 
+/**
+ * Small helper function to create a map using the Map constructor arguments.
+ * This is needed because IE11 does not implement that constructor.
+ */
+function makeMap<K, V>(values: Array<[K, V]>): Map<K, V> {
+  const map: Map<K, V> = new Map();
+  values.forEach(([k, v]) => {
+    map.set(k, v);
+  });
+  return map;
+}
+
 /**
  * Types of raw signals from the browser caught at the top level.
  *
  * For events like 'submit' or audio/video events which don't consistently
  * bubble (which we trap at a lower node than `document`), binding
  * at `document` would cause duplicate events so we don't include them here.
  */
-export const topLevelTypes = {
-  topAnimationEnd: getVendorPrefixedEventName('animationend'),
-  topAnimationIteration: getVendorPrefixedEventName('animationiteration'),
-  topAnimationStart: getVendorPrefixedEventName('animationstart'),
-  topBlur: 'blur',
-  topCancel: 'cancel',
-  topChange: 'change',
-  topClick: 'click',
-  topClose: 'close',
-  topCompositionEnd: 'compositionend',
-  topCompositionStart: 'compositionstart',
-  topCompositionUpdate: 'compositionupdate',
-  topContextMenu: 'contextmenu',
-  topCopy: 'copy',
-  topCut: 'cut',
-  topDoubleClick: 'dblclick',
-  topDrag: 'drag',
-  topDragEnd: 'dragend',
-  topDragEnter: 'dragenter',
-  topDragExit: 'dragexit',
-  topDragLeave: 'dragleave',
-  topDragOver: 'dragover',
-  topDragStart: 'dragstart',
-  topDrop: 'drop',
-  topFocus: 'focus',
-  topInput: 'input',
-  topKeyDown: 'keydown',
-  topKeyPress: 'keypress',
-  topKeyUp: 'keyup',
-  topLoad: 'load',
-  topLoadStart: 'loadstart',
-  topMouseDown: 'mousedown',
-  topMouseMove: 'mousemove',
-  topMouseOut: 'mouseout',
-  topMouseOver: 'mouseover',
-  topMouseUp: 'mouseup',
-  topPaste: 'paste',
-  topScroll: 'scroll',
-  topSelectionChange: 'selectionchange',
-  topTextInput: 'textInput',
-  topToggle: 'toggle',
-  topTouchCancel: 'touchcancel',
-  topTouchEnd: 'touchend',
-  topTouchMove: 'touchmove',
-  topTouchStart: 'touchstart',
-  topTransitionEnd: getVendorPrefixedEventName('transitionend'),
-  topWheel: 'wheel',
-};
+export const topLevelTypes: Map<TopLevelType, string> = makeMap([
```

Comment:
Just an idea: since we have to map the event types to the native event names anyways, I wonder if we can just use the event names as the event type constants? Then we could potentially remove the indirection of mapping event types to event names.

I think we'd have to ensure that GCC didn't inline the strings, and I'm not sure if we can control that with current `closure-compiler-js` API.

Reply:
> Just an idea: since we have to map the event types to the native event names anyways, I wonder if we can just use the event names as the event type constants? Then we could potentially remove the indirection of mapping event types to event names.

I had to try that. Please have a look at my latest commit and the updated bundle sizes :)

> I think we'd have to ensure that GCC didn't inline the strings, and I'm not sure if we can control that with current closure-compiler-js API.

Even if we don't care about GCC inlining, using the raw event names still significantly reduces the bundle size while avoiding `Map` completely (might still be in my WIP commit but can definitely be removed).

Compared to the number approach, the gzipped version is even smaller! :) What do y'all think about that?

Reply:
> Compared to the number approach, the gzipped version is even smaller! 

At least when using the build numbers of my previous commit as a baseline, this was the result. On CI, the gzipped file sizes seems to be pretty much identical.

**Local Builds** 

Prev Size: 98.7 KB 
Current Size: 99.08 KB 
Diff: +0.4%

Prev Gzip: 31.73 KB
Current Gzip: 31.73 KB
Diff: -0.2%

**CI Sizes**

- Number IDs: react-dom.production.min.js	-1.6%	-1.0%	100.2 KB	98.63 KB	31.97 KB	31.66 KB	UMD_PROD
- Raw Event Name IDs: react-dom.production.min.js	-1.1%	-1.0%	100.2 KB	99.08 KB	31.97 KB	31.67 KB	UMD_PROD